sdk/src/main/scala/com/horizen/account/api/rpc/service/EthService.scala

```diff
@@ -786,6 +794,44 @@ class EthService(
         // apply requested transaction with tracing enabled
         tagStateView.applyTransaction(requestedTx, previousTransactions.length, gasPool, blockContext)
 
+        // return the tracer result from the evm
+        if(blockContext.getEvmResult != null) {
+          blockContext.getEvmResult.tracerResult
+        } else Unit
+      }
+    }
+  }
+
+  @RpcMethod("debug_traceCall")
+  @RpcOptionalParameters(1)
+  def traceCall(params: TransactionArgs, tag: String, config: TraceOptions): Any = {
+
+    applyOnAccountView { nodeView =>
+      // get block to trace
+      val (block, blockInfo) = getBlockById(nodeView, getBlockIdByHashOrTag(nodeView, tag))
+
+      // get state at selected block
+      getStateViewAtTag(nodeView, if(tag=="pending") "pending" else (blockInfo.height).toString) { (tagStateView, blockContext) =>
+
+        // use default trace params if none are given
+        blockContext.setTraceParams(if (config == null) new TraceOptions() else config)
+
+        // apply mainchain references
+        for (mcBlockRefData <- block.mainchainBlockReferencesData) {
+          tagStateView.applyMainchainBlockReferenceData(mcBlockRefData)
+        }
+
+        val gasPool = new GasPool(block.header.gasLimit)
+
+        // apply block transactions without tracing
+        for ((tx, i) <- block.transactions.zipWithIndex) {
+          tagStateView.applyTransaction(tx, i, gasPool, blockContext)
+        }
```

Comment:
Previously you we're opening the state after the parent block and then reapplying the transactions of this block. Now you open the state at the desired block - so that already includes the transactions of the block.
I think we can remove this - it should fail anyway because the nonces are not valid anymore.

Reply:
Agree, the previous implementation was not correct. Removed